src/healthcheck.rs

```diff
@@ -0,0 +1,27 @@
+use std::error::Error;
+use std::net::SocketAddr;
+use tokio::io::AsyncWriteExt;
+use tokio::net::TcpListener;
+
+#[tracing::instrument]
+pub async fn build_healthcheck_server() -> Result<(), Box<dyn Error>> {
```

Comment:
i think this should be renamed, this function also runs the server, doesnt just start it

Reply:
Yep renamed both